### PR TITLE
Use message link security attribute to also control link security on transmission

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Comment by jwhui** [comment]
@darconeous

File: src/core/common/message.cpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ ThreadError Message::Init(void)
     return kThreadError_None;
 }
 
-Message *Message::New(uint8_t aType, uint16_t aReserved)
+Message *Message::New(uint8_t aType, uint16_t aReserved, bool aLinkSecurityEnabled)
```
Comment (darconeous) [inline src/core/common/message.cpp]: Why add this here? Setting the link security with `SetLinkSecurity()` after it's been allocated doesn't seem unreasonable.

File: src/core/common/message.cpp [hunk 2]
```diff
@@ -498,14 +499,14 @@ void Message::SetDirectTransmission(void)
     mInfo.mDirectTx = true;
 }
 
-bool Message::GetSecurityValid(void) const
+bool Message::IsLinkSecurityEnabled(void) const
```
Comment (darconeous) [inline src/core/common/message.cpp]: 👍  on the new name.

**Comment by darconeous** [comment]
Overall, this looks good to me. 👍

File: src/core/common/message.cpp (hunk 1 above)
Comment (jwhui) [inline src/core/common/message.cpp]: I thought link security was important enough to force the implementor to think about it when allocating a new message.  However, I would also be fine with having link security always `true` by default and removing the required argument from `Message::New`.  Would you prefer that?

Comment (darconeous) [inline src/core/common/message.cpp]: I think that would be preferable. That way the user always defaults to sending secure packets and must opt-in to sending insecure packets.

Comment (jwhui) [inline src/core/common/message.cpp]: Sounds good to me.  I just updated the PR to match what we discussed.

**Comment by darconeous** [comment]
👍

**Pull request merged**